components/widget/widget_factory.lua

```diff
@@ -0,0 +1,23 @@
+---
+-- @Liquipedia
+-- wiki=commons
+-- page=Module:Widget/Factory
+--
+-- Please see https://github.com/Liquipedia/Lua-Modules to contribute
+--
+
+local Class = require('Module:Class')
+
+local WidgetFactory = {}
+
+---@param args {widget: string, children: ((Widget|Html|string|number)[])|Widget|Html|string|number, [any]:any}
+---@return Widget
+function WidgetFactory.fromTemplate(args)
+	local widgetClass = args.widget
+	args.widget = nil
+	args.children = type(args.children) == 'table' and args.children or {args.children}
+	local WidgetClass = require('Module:Widget/' .. widgetClass)
```

Comment:
possibly lua import instead?
also would we want to do an exist check on the module?

Reply:
Changed to lua import. 

Don't really see the need of checking. It's either gonna error here or on the next line where there kinda needs to be an asset to verify that it exists